src/transport/assembly_dg.hh

```diff
@@ -0,0 +1,928 @@
+
+                        for (unsigned int i=0; i<n_dofs[n]; i++)
+                            for (int m=0; m<2; m++)
+                                for (unsigned int j=0; j<n_dofs[m]; j++)
+                                    local_matrix_[(i+n*n_dofs[0])*(n_dofs[0]+n_dofs[1]) + m*n_dofs[0] + j] +=
+                                            comm_flux[m][n]*fv_sb_[m]->shape_value(j,k)*fv_sb_[n]->shape_value(i,k);
+                    }
+                }
+                data_->ls[sbi]->mat_set_values(n_dofs[0]+n_dofs[1], &(side_dof_indices_vb_[0]), n_dofs[0]+n_dofs[1], &(side_dof_indices_vb_[0]), &(local_matrix_[0]));
+            }
+        }
+
+    }
+
+
+    /// Assembles the right hand side vector due to volume sources.
+    void set_sources(DHCellAccessor cell) override
+    {
+    	ASSERT_EQ_DBG(cell.dim(), dim).error("Dimension of element mismatch!");
+
+        ElementAccessor<3> elm = cell.elm();
+
+        fe_values_.reinit(elm);
+        cell.get_dof_indices(dof_indices_);
+        cell.get_loc_dof_indices(loc_dof_indices_);
+
+        model_.compute_source_coefficients(fe_values_.point_list(), elm, sources_conc_, sources_density_, sources_sigma_);
+
+        // assemble the local stiffness matrix
+        for (unsigned int sbi=0; sbi<model_.n_substances(); sbi++)
+        {
+            fill_n( &(local_rhs_[0]), ndofs_, 0 );
+            local_source_balance_vector_.assign(ndofs_, 0);
+            local_source_balance_rhs_.assign(ndofs_, 0);
+
+            // compute sources
+            for (unsigned int k=0; k<qsize_; k++)
+            {
+                source = (sources_density_[sbi][k] + sources_conc_[sbi][k]*sources_sigma_[sbi][k])*fe_values_.JxW(k);
+
+                for (unsigned int i=0; i<ndofs_; i++)
+                    local_rhs_[i] += source*fe_values_.shape_value(i,k);
+            }
+            data_->ls[sbi]->rhs_set_values(ndofs_, &(dof_indices_[0]), &(local_rhs_[0]));
+
+            for (unsigned int i=0; i<ndofs_; i++)
+            {
+                for (unsigned int k=0; k<qsize_; k++)
+                    local_source_balance_vector_[i] -= sources_sigma_[sbi][k]*fe_values_.shape_value(i,k)*fe_values_.JxW(k);
+
+                local_source_balance_rhs_[i] += local_rhs_[i];
+            }
+            model_.balance()->add_source_values(model_.get_subst_idx()[sbi], elm.region().bulk_idx(), loc_dof_indices_,
+                                               local_source_balance_vector_, local_source_balance_rhs_);
+        }
+    }
+
+
+    void set_boundary_conditions(DHCellAccessor cell) override
+    {
+        if (cell.elm()->boundary_idx_ == nullptr) return;
+
+        for (unsigned int si=0; si<cell.elm()->n_sides(); si++)
+        {
+            const Edge *edg = cell.elm().side(si)->edge();
+            if (edg->n_sides > 1) continue;
+            // skip edges lying not on the boundary
+            if (edg->side(0)->cond() == NULL) continue;
+
+            if (edg->side(0)->dim() != dim-1)
```

Comment:
This can be removed. Must test it on a problem with DG on mixed dimension (26/01).